terraform/modules/scheduled-function/main.tf

```diff
@@ -0,0 +1,169 @@
+# Scheduled Cloud Function Module
+# This module creates a complete scheduled Cloud Function setup with all necessary components
+
+# Required providers
+terraform {
+  required_version = ">= 1.3.0"
+
+  required_providers {
+    google = {
+      source  = "hashicorp/google"
+      version = ">= 6.0.0"
+    }
+    archive = {
+      source  = "hashicorp/archive"
+      version = ">= 2.0.0"
+    }
+    null = {
+      source  = "hashicorp/null"
+      version = ">= 3.0.0"
+    }
+  }
+}
+
+# Service account for the Cloud Function
+resource "google_service_account" "function_sa" {
+  project      = var.project_id
+  account_id   = "${var.function_name}-sa"
+  display_name = "Service Account for ${var.function_name} function"
+  description  = "Service account used by the ${var.function_name} scheduled Cloud Function"
+}
+
+# Storage bucket for function source code
+resource "google_storage_bucket" "function_bucket" {
+  project                     = var.project_id
+  name                        = "${var.function_name}-source-${var.project_id}"
+  location                    = var.region
+  uniform_bucket_level_access = true
+
+  # Optional: Add lifecycle rules to clean up old versions
+  lifecycle_rule {
+    condition {
+      age = 7 # Keep old versions for 7 days
+    }
+    action {
+      type = "Delete"
+    }
+  }
+}
+
+# Install dependencies (if requirements file exists)
+resource "null_resource" "install_dependencies" {
+  count = fileexists("${var.source_dir}/${var.requirements_file}") ? 1 : 0
+
+  triggers = {
+    # Re-run if requirements file changes
+    requirements_hash = filemd5("${var.source_dir}/${var.requirements_file}")
+    # Re-run if source files change
```

Comment:
Why do you have to reinstall dependencies if the source files change?  I'd think you'd just care if the requirements.txt file changed.